discovery/gossiper.go

```diff
@@ -1616,17 +1773,25 @@ func (d *AuthenticatedGossiper) processNetworkAnnouncement(
 		// newer update for this node so we can skip validating
 		// signatures if not required.
 		if d.cfg.Router.IsStaleNode(msg.NodeID, timestamp) {
+			log.Debugf("Skipped processing stale node: %x",
+				msg.NodeID)
 			nMsg.err <- nil
 			return nil, true
 		}
 
 		if err := d.addNode(msg, schedulerOp...); err != nil {
-			if routing.IsError(err, routing.ErrOutdated,
-				routing.ErrIgnored) {
+			log.Debugf("Adding node: %x got error: %x",
+				msg.NodeID, err)
 
-				log.Debug(err)
-			} else if err != routing.ErrVBarrierShuttingDown {
+			switch {
+			case routing.IsError(err, routing.ErrOutdated,
+				routing.ErrIgnored):
+
+			case err == routing.ErrVBarrierShuttingDown:
```

Comment:
nit: maybe it's worth to add `ErrVBarrierShuttingDown` and `ErrParentValidationFailed` to the `routing/errors.go` iota error codes and collapse this two cases in one.

Reply:
good point

Reply:
Cool added a commit to switch the error codes.